tests/system_tests/test_variables.py

```diff
@@ -0,0 +1,182 @@
+import os
+import os.path
+from textwrap import dedent
+
+import ptvsd
+from ptvsd.wrapper import INITIALIZE_RESPONSE  # noqa
+from tests.helpers.debugclient import EasyDebugClient as DebugClient
+from tests.helpers.debugsession import Awaitable
+
+from . import (
+    LifecycleTestsBase,
+    lifecycle_handshake,
+)
+
+
+ROOT = os.path.dirname(os.path.dirname(ptvsd.__file__))
+PORT = 9876
+CONNECT_TIMEOUT = 3.0
+
+
+class VariableLifecycleTests(LifecycleTestsBase):
+
+    def test_variables(self):
+        source = dedent("""
+            a = 1
+            b = {"one":1, "two":2}
+            # <Token>
+            c = 3
+            """)
+        (filename, filepath, env, expected_module, argv,
+         cwd) = self.get_test_info(source)
+
+        bp_line = self.find_line(filepath, 'Token')
+        breakpoints = [{
+            "source": {
+                "path": filepath
+            },
+            "breakpoints": [{
+                "line": bp_line
+            }]
+        }]
+
+        with DebugClient(port=PORT, connecttimeout=CONNECT_TIMEOUT) as editor:
+            adapter, session = editor.host_local_debugger(
+                argv, env=env, cwd=cwd, timeout=CONNECT_TIMEOUT)
+
+            terminated = session.get_awaiter_for_event('terminated')
+            exited = session.get_awaiter_for_event('exited')
+            thread_exit = session.get_awaiter_for_event('thread', lambda msg: msg.body.get("reason", "") == "exited") # noqa
+
+            with session.wait_for_event("stopped") as result:
+                (
+                    req_initialize,
+                    req_launch,
+                    req_config,
+                    reqs_bps,
+                    _,
+                    _,
+                ) = lifecycle_handshake(
+                    session, "launch", breakpoints=breakpoints)
+            req_bps, = reqs_bps  # There should only be one.
+            tid = result["msg"].body["threadId"]
+
+            stacktrace = session.send_request("stackTrace", threadId=tid)
+            stacktrace.wait()
+            frame_id = stacktrace.resp.body["stackFrames"][0]["id"]
+            scopes = session.send_request('scopes', frameId=frame_id)
+            scopes.wait()
+            variables_reference = scopes.resp.body["scopes"][0]["variablesReference"] # noqa
+            variables = session.send_request('variables', variablesReference=variables_reference) # noqa
+            variables.wait()
+
```

Comment:
Can you ensure that the variables are sorted? We also need to ensure that sort order. Like a, b, c, before _a,_b,_c before __a,__b,__c before __a__,__b__,__c__.

Reply:
Will create a separate test for that.
Else this test gets overloaded and difficult to maintain.

Reply:
@karthiknadig 
Apart from that, is this OK